ref/general/hardening-intro.adoc

```diff
@@ -22,15 +22,16 @@ Hardening is a general term that is used to describe the measures that are taken
 
-To learn more about operating system vulnerabilities and ways to prevent operating system intrusions, go to <link pending>.
+To learn more about OS vulnerabilities and ways to prevent OS intrusions, go to <link pending>.
 
 ==== Best practices
 
-* *Keep Open Liberty up to date*: To reduce operating system vulnerabilities, it's important to keep your version of Open Liberty up to date. Open Liberty fix packs are compatible with later versions, and the zero migration policy means that you can move to a newer fix pack without changing anything in your server configuration. Keeping Open Liberty up to date helps ensure that you have the most recent link:/docs/ref/general/#security-vulnerabilities.html[critical security fixes]. To get the latest version of Open Liberty, go to the link:https://openliberty.io/downloads/[downloads page].
+* *Keep Open Liberty up to date*: To reduce OS vulnerabilities, it's important to keep your version of Open Liberty up to date. Open Liberty fix packs are compatible with later versions, and the zero migration policy means that you can move to a newer fix pack without changing anything in your server configuration. Keeping Open Liberty up to date helps ensure that you have the most recent link:/docs/ref/general/#security-vulnerabilities.html[critical security fixes]. To get the latest version of Open Liberty, go to the link:https://openliberty.io/downloads/[downloads page].
 * *Use user registries that are safe for production*: Basic and Quick Start Security user registries are useful for development environments, but they are not acceptable for readiness testing, QA, or production use. Instead, use an LDAP user registry, federated repositories, or create a custom user registry. If you use an LDAP user registry, maintain your bind password in a separate `include` file, not in the `server.xml` file. If you use a custom registry, use the `UserRegistry` class to implement the registry, and make sure that the registry is thoroughly reviewed before you use it in production.
-* *Carefully configure file permissions*: When you configure file permissions, ensure that the users who want access to the configuration fall into the appropriate permission bit. Users that are not the owner and are not part of the authorized group fall into the `other` permission bit, which is marked as `0`.
+* *Use web single-sign on*: For user authentication, it is common to use single-sign on (SSO) technologies (such as SAML, JWT, or OpenID Connect) instead of direct user registry access from Open Liberty. SSO can be used instead of user registries, and it provides better security and regulatory compliance. If you use SSO, maintain your SSO configuration in a separate include file, not in the `server.xml` file. For more information about configuring SSO, go to <link pending>.
```

Comment:
"instead of direct user registry access from Open Liberty." can be rephrased as "instead of directly accessing user registry from Open Liberty" for clarity.

"SSO can be used instead of user registries, and it provides better security and regulatory compliance." repeats the previous sentence. You can say, "SSO  provides better security and regulatory compliance. "

Reply:
Updated